src/client/Microsoft.Identity.Client/Internal/Requests/Interactive/InteractiveRequest.cs

```diff
@@ -0,0 +1,170 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client.ApiConfig.Parameters;
+using Microsoft.Identity.Client.Core;
+using Microsoft.Identity.Client.Internal.Broker;
+using Microsoft.Identity.Client.OAuth2;
+using Microsoft.Identity.Client.TelemetryCore.Internal;
+using Microsoft.Identity.Client.TelemetryCore.Internal.Events;
+using Microsoft.Identity.Client.UI;
+
+namespace Microsoft.Identity.Client.Internal.Requests
+{
+
+    /// <summary>
+    /// This class decides the workflow of an interactive request. The business rules are: 
+    /// 
+    /// 1. If WithBroker is set to true
+    /// 1.1. Attempt to invoke the broker and get the token
+    /// 1.2. If this fails, e.g. if broker is not installed, the use a web view (goto 2)
+    /// 
+    /// 2. Use a webview and get an auth code and look at the auth code
+    /// 2.1. If the auth code has a special format, showing that a broker is needed then. Invoke the broker flow (step 1) with a broker installation url
+    /// 2.2. Otherwise exchange the auth code for tokens (normal authorize_code grant)
+    /// </summary>
+    internal class InteractiveRequest : RequestBase
```

Comment:
RequestBase is the only place that creates an ApiEvent. Currently there is only 1 api event per flow, and I did not want to reactor telememetry. So only top level request inherits requestbase.

PS: inheritance sucks.